testing-suite/staking-v4/scenarios/PR_6114.py

```diff
@@ -0,0 +1,181 @@
+# Unstake 1 or more eligible/waiting nodes (no more than all nodes per shard or no more than all eligible)
+# and call auction list api; we should see 8 qualified nodes. Next epoch we need to check that exactly 8 nodes were qualified.
+# Afterwards(some epochs), those leaving nodes will be leaving and replaced by the auction nodes
+
+
+# Steps:
+#
+# 1) In epoch 5 stake with A that 2 keys  - will go in auction
+# 2) Add blocks and epochs until 1 of this 2 keys are eligible - let's call it epoch X
+# 3) unStake the eligible key in epoch X
+# 4) call auction list api after 2 blocks (we are still in epoch X) and:
+#       4.1) unStaked key status should be "eligible (leaving)" on validator/statistics
+#       4.2) we should have still 8 keys qualified
+#       4.3) we should have still 2 keys not qualified
+#       4.4) we should have 39 keys with status "eligible"
+#       4.5) we should have still 16 keys with status "waiting"
+# 5) go to epoch X+1  and 5 blocks and make sure:
+#       5.1) unstaked key should be inactive
+#       5.2) 8 keys from auction (qualified or not qualified , cause is not deterministic)  are now waiting
+#       5.3) we still have 9 keys in auction, 8 of them are qualified, 1 not qualified , 40 keys eligible, 16 waiting
+
+
+def main():
+    print("Happy testing")
+
+
+def test_PR_6114():
+
+    # === PRE-CONDITIONS ==============================================================
+    AMOUNT_TO_MINT = "10000" + "000000000000000000"
+
+    _A = Wallet(Path("./wallets/walletKey_1.pem"))
+    key_1 = ValidatorKey(Path("./validatorKeys/validatorKey_1.pem"))
+    key_2 = ValidatorKey(Path("./validatorKeys/validatorKey_2.pem"))
+    all_keys = [key_1, key_2]
+
+    # check if minting is successful
+    assert "success" in _A.set_balance(AMOUNT_TO_MINT)
+
+    # add some blocks
+    response = add_blocks(5)
+    assert "success" in response
+    time.sleep(0.5)
+
+    # check balance
+    assert _A.get_balance() == AMOUNT_TO_MINT
+
+    # add keys to protocol. This way staked keys will not go to jail.
+    assert "success" in add_key(all_keys)
+
+    # move to epoch
+    assert "success" in add_blocks_until_epoch_reached(5)
+
+    # === STEP 1 ==============================================================
+    # 1) In epoch 5 stake with A that 2 keys  - will go in auction
+
+    tx_hash = stake(_A, all_keys)
+
+    # check if tx is success
+    assert "success" in add_blocks_until_tx_fully_executed(tx_hash)
+
+    # check that nodes are staked
+    for key in all_keys:
+        assert "staked" in key.get_status(_A.public_address())
+
+    # move 2 blocks before calling validator/statistics
+    assert "success" in add_blocks(2)
+    time.sleep(1)
+
+    # check that nodes are in auction
+    for key in all_keys:
+        assert "auction" in key.get_state()
+
+    # === STEP 2 ==============================================================
+    # 2) Add blocks and epochs until 1 of this 2 keys are eligible - let's call it epoch X
```

Comment:
Better here would actually be to wait until:
- at least one of his key is eligibile
- and the other one is either waiting/eligibile.

Then unstake both nodes and check new results. Anyway, can be kept as it is.

Reply:
This particular scenario is done with 1 unstaked key , so we will keep it as it is and I will create another test / scenario with unstaking one eligible and one waiting in another PR.